applications/MeshMovingApplication/tests/cpp_tests/test_mesh_velocity_calculation.cpp

```diff
@@ -0,0 +1,230 @@
+        }
+    }
+}
+
+template<class TBDF>
+void TestBDFTimeIntegration(const TBDF& rBDF,
+                            const ResultsMapType& rResultsX,
+                            const ResultsMapType& rResultsY,
+                            const bool SetupTest)
+{
+    Model current_model;
+    const std::size_t buffer_size = TimeDiscretization::GetMinimumBufferSize(rBDF);
+    const std::size_t num_steps = 3;
+    const double delta_time = 0.1;
+    const bool requires_mesh_acceleration = false;
+
+    CreateModelPartForMeshVelComputation(current_model, buffer_size, delta_time, requires_mesh_acceleration);
+
+    auto& r_model_part = current_model.GetModelPart("MeshVelMP");
+
+    // pseudo time-loop to check results
+    const double current_time = r_model_part.GetProcessInfo()[TIME];
+    for (std::size_t i_step = 0; i_step < num_steps; ++i_step){
+        r_model_part.CloneTimeStep(current_time + (i_step+1) * delta_time);
+        array_1d<double,3> u_val = ZeroVector(3);
+        for (auto it_node : r_model_part.NodesArray()) {
+            // apply some deformation
+            u_val(0) = 2.0*std::pow(i_step, 1.82) * delta_time * it_node->X();
+            u_val(1) = 1.0*std::pow(i_step, 3.951) * delta_time * it_node->Y();
+            it_node->GetSolutionStepValue(MESH_DISPLACEMENT) = u_val;
+        }
+        MeshVelocityCalculation::CalculateMeshVelocities(r_model_part, rBDF);
+
+        CheckNodalResults(r_model_part, rResultsX, MESH_VELOCITY_X, i_step, SetupTest);
+        CheckNodalResults(r_model_part, rResultsY, MESH_VELOCITY_Y, i_step, SetupTest);
+    }
+}
+
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateMeshVelocitiesBDF1, MeshMovingApplicationFastSuite)
+{
+    const ResultsMapType results_x {
+        {5 ,  {-0.5, 0.66666666666, 1.68720799011}},
+        {13 , {-1.5, 2.0,           5.06162397033}},
+    };
+
+    const ResultsMapType results_y {
+        {2 ,  {-1.0, 0.33333333333333, 4.82189918435}},
+        {3 ,  {-2.0, 0.66666666666,    9.64379836869}},
+    };
+
+    TimeDiscretization::BDF1 bdf;
+
+    TestBDFTimeIntegration(bdf, results_x, results_y, false);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateMeshVelocitiesBDF2, MeshMovingApplicationFastSuite)
+{
+    const ResultsMapType results_x {
+        {5 ,  {-1.08333333333, 1.3333333333333, 2.19747865183}},
+        {13 , {-3.25,          4.0,             6.59243595549}},
+    };
+
+    const ResultsMapType results_y {
+        {2 ,  {-2.16666666667, 1.16666666667, 7.06618210985}},
+        {3 ,  {-4.33333333333, 2.33333333333, 14.1323642197}},
+    };
+
+    TimeDiscretization::BDF2 bdf;
+
+    TestBDFTimeIntegration(bdf, results_x, results_y, false);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateMeshVelocitiesGeneralizedAlpha, MeshMovingApplicationFastSuite)
+{
+    TimeDiscretization::GeneralizedAlpha gen_alpha(-0.05, 0.03);
+    Model current_model;
+    const std::size_t buffer_size = TimeDiscretization::GetMinimumBufferSize(gen_alpha);
```

Comment:
Simpler
```suggestion
    const std::size_t buffer_size = GetMinimumBufferSize(gen_alpha);
```

Reply:
this works?
I will try

Reply:
yes, it should: https://en.cppreference.com/w/cpp/language/adl